src/cwe_checker_lib/src/ghidra_pcode/pcode_op_simple/jumps/mod.rs

```diff
@@ -0,0 +1,152 @@
+    ///
+    /// The target is either a pcode operation relative target, or an absolute machine instruction target.
+    pub fn get_jump_target(&self) -> Option<JmpTarget> {
+        use crate::pcode::JmpType::*;
+        if let PcodeOperation::JmpType(jmp_type) = &self.pcode_mnemonic {
+            // Pcode definition distinguishes between `location` and `offset`.
+            // Note: $(GHIDRA_PATH)/docs/languages/html/pcodedescription.html#cpui_branch
+            // Currently, the IR does not distinguishes these cases.
+            // We do nothing here.
+            match jmp_type {
+                BRANCH | CBRANCH | CALL => (),                  // case `location`
+                BRANCHIND | CALLIND | CALLOTHER | RETURN => (), // case `offset`
+            }
+            if let Some(target) = self.input0.get_ram_address() {
+                return Some(JmpTarget::Absolute(target.try_to_u64().unwrap()));
+            } else if let Expression::Const(jmp_offset) = self.input0.into_ir_expr().unwrap() {
+                if let Some(target_index) = self
+                    .pcode_index
+                    .checked_add_signed(jmp_offset.try_to_i64().unwrap())
+                {
+                    return Some(JmpTarget::Relative((self.pcode_index, target_index)));
+                } else {
+                    // TODO: Negative target index, trigger log message here
+                    return Some(JmpTarget::Relative((self.pcode_index, 0)));
+                }
+            }
+        }
+        None
+    }
+
+    pub fn into_ir_jump(&self, address: &String) -> Term<Jmp> {
+        if let PcodeOperation::JmpType(a) = self.pcode_mnemonic {
+            match a {
+                JmpType::BRANCH => self.create_branch(address),
+                JmpType::CBRANCH => self.create_cbranch(address),
+                JmpType::BRANCHIND => self.create_branch_indirect(address),
+                JmpType::CALL => self.create_call(address),
+                JmpType::CALLIND => self.create_call_indirect(address),
+                JmpType::CALLOTHER => self.create_call_other(address),
+                JmpType::RETURN => self.create_return(address),
+            }
+        } else {
+            panic!("Not a jump operation")
+        }
+    }
+
+    /// Determines the target ad returns corresponding `Tid`.
+    ///
+    /// Pcode relative targets are prefixed with `artificial`, e.g. artificial_blk_0x100_4.
+    /// Absolute targets (RAM located) are not prefixed, e.g. blk_0x0200
+    fn extract_target(&self, address: &String) -> Tid {
+        let (id, address) = match self.get_jump_target() {
+            Some(JmpTarget::Absolute(_)) => {
+                (format!("blk_{}", self.input0.id), self.input0.id.clone())
+            }
+            Some(JmpTarget::Relative((_, target_index))) => (
+                format!("artificial_blk_{}_{}", address, target_index),
+                address.clone(),
+            ),
```

Comment:
Here is a good place to see the problem with the `artificial` prefix. Absolute targets can also be artificial in the sense that Ghidra did not correctly split a block at the target address and we have to split the block at the target address ourselves.